lib/jellygrinder/ll_client.ex

```diff
@@ -46,25 +48,22 @@ defmodule Jellygrinder.LLClient do
 
   @impl true
   def handle_info(:get_new_partials, state) do
+    path = Path.join(state.base_path, state.track_manifest_name)
+    query = create_track_manifest_query(state)
+
     latest_partial =
-      case request_track_manifest(state) do
+      case request(path <> query, "media playlist", state) do
         {:ok, track_manifest} ->
           track_manifest
           |> get_new_partials_info(state.latest_partial)
-          |> Enum.reduce(state.latest_partial, fn partial, _acc ->
-            request_partial(partial, state)
-          end)
+          |> Enum.map(&request_partial(&1, state))
+          |> List.last(state.latest_partial)
 
         {:error, _response} ->
           state.latest_partial
       end
 
-    wait_for =
-      if is_nil(latest_partial),
-        do: @default_wait_period,
-        else: floor(latest_partial.duration * 1000)
-
-    Process.send_after(self(), :get_new_partials, wait_for)
+    send(self(), :get_new_partials)
```

Comment:
What do you think about creating an API to communicate with this GenServer? Where we would use `GenServer.cast`, and these casts would be wrapped in public functions.

Reply:
Why do we need the functions to be public? We don't expect any other module to call them

Reply:
Ok in this case it could be a private function but as we use it in two places I still would move it to a function.